### PR TITLE
tests: enable e2e, retries, bug fixes

### PR DESCRIPTION
**Pull request by hayemaxi**
https://github.com/aws/aws-toolkit-vscode/issues/3339
Failures for windows insiders will cancel stable test.

For e2e tests, will we monitor for any throttling.

Also:
- Retry strategy chosen somewhat arbitrarily, but these settings don't seem to delay the test suite by much if at all. In recent runs with these, throttling has stopped.

- Adds retries with jitter by utilizing retries in the SDK, see
https://github.com/aws/aws-sdk-js/blob/3e616251947c73d5239178c167a9d73d985ca581/lib/util.js#L884.

- Also fixes various bugs with the tests that causes errors to be thrown inadvertently, including type checking/verifty the code of some
exceptions. In one case, this threw an AccessDeniedException when checking a dev env that was already deleted previously.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by hayemaxi** [comment]
throttling with no parallel runs :(

File: src/testE2E/codecatalyst/client.test.ts [hunk 1]
```diff
@@ -354,16 +355,15 @@ describe.skip('Test how this codebase uses the CodeCatalyst API', function () {
             assert.ok(spaces.find(space => space.name === spaceName))
         })
 
-        it('lists all projects for the given user', async function () {
+        // TODO: Re-add this test and delete the projects after the test suite with the appropriate SDK API.
+        it.skip('lists all projects for the given user', async function () {
```
Comment (justinmk3) [inline src/testE2E/codecatalyst/client.test.ts:359]: Assuming the skip() is still intended, I don't quite understand the comment. What's meant by "the appropriate sdk api" ?

**Review by justinmk3 — APPROVED** [review]

File: src/testE2E/codecatalyst/client.test.ts (hunk 1 above)
Comment (hayemaxi) [reply src/testE2E/codecatalyst/client.test.ts:359]: updated this comment and one other one

**Review by justinmk3 — APPROVED** [review]

**Comment by nkomonen-amazon** [comment]
Nice! @hayemaxi one thing I learned recently is that editing the `/runIntegrationTests` comment and adding some spaces will trigger the integration tests so you don't need to make a new comment each time.

**Review by nkomonen-amazon — APPROVED** [review]

**Pull request merged**